src/lock_file.cpp

```diff
@@ -89,6 +90,12 @@ LockFile::LockFile(const char* path_, PassKey) : path(path_)
 
 LockFile& LockFile::Get(const char* path)
 {
+#ifdef _WIN32
+    // The character ':' is reserved on Windows and cannot be used for constructing
+    // a path except when it follows a drive letter.
+    std::string file{std::regex_replace(path, std::regex(":memory:"), "memory_")};
+    path = file.c_str();
```

Comment:
[Question] @JehandadKhan @Dmantri98 Is is related to the memvfs used to hold SQLite databases in memory? And we use normal FS for locking?

Reply:
Yes, that is related to the memvfs, and we use normal FS locking. I am not super happy with this fix. If you think it is not required, I would gladly replace it with something else, like not locking when working with an in-memory database.

Reply:
IIUC the special path `:memory:` never results in file IO because sqlite would consume it completely. @apwojcik  Are you saying that the sqlite behavior on windows is different. 

Keep in mind that even on Linux a file name `:memory:` would not be created.

Reply:
> IIUC the special path `:memory:` never results in file IO because sqlite would consume it completely. @apwojcik Are you saying that the sqlite behavior on windows is different.
> 
> Keep in mind that even on Linux a file name `:memory:` would not be created.

@JehandadKhan On Windows, it is trying to create such a path. That is the reason why we came up with this PR.